src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearchWidget.ts

```diff
@@ -0,0 +1,276 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { localize } from 'vs/nls';
+import { IKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { Widget } from 'vs/base/browser/ui/widget';
+import { FindInput, IFindInputOptions } from 'vs/base/browser/ui/findinput/findInput';
+import { IFocusTracker, append, $, trackFocus } from 'vs/base/browser/dom';
+import { IContextKey, IContextKeyService } from 'vs/platform/contextkey/common/contextkey';
+import { Emitter, Event } from 'vs/base/common/event';
+import { IContextViewService } from 'vs/platform/contextview/browser/contextView';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { IKeybindingService } from 'vs/platform/keybinding/common/keybinding';
+import { IClipboardService } from 'vs/platform/clipboard/common/clipboardService';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IAccessibilityService } from 'vs/platform/accessibility/common/accessibility';
+import * as Constants from 'sql/workbench/contrib/notebook/common/constants';
+import { IMessage } from 'vs/base/browser/ui/inputbox/inputBox';
+import { appendKeyBindingLabel } from 'vs/workbench/contrib/search/browser/searchActions';
+import { ContextScopedFindInput } from 'vs/platform/browser/contextScopedHistoryWidget';
+import { attachFindReplaceInputBoxStyler } from 'vs/platform/theme/common/styler';
+import { isMacintosh } from 'vs/base/common/platform';
+import { KeyMod, KeyCode } from 'vs/base/common/keyCodes';
+
+export interface INotebookExplorerSearchOptions {
```

Comment:
Can we reuse ISearchWidgetOptions instead? I realize there are extra optional fields, but I really want to get out of the business of duplicating as little code as possible.

Reply:
I'll clean this up in a separate PR with my extensible changes.